### PR TITLE
Bug: itertools.pairwise in Python < 3.10

### PR DESCRIPTION
**Pull request by ilya-causalens**
## Description and Motivation

- Fixed bug where use of `itertools.pairwise` would result in errors for Python \< 3.10.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [ ] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [ ] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [ ] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

**Review by steven-causalens — APPROVED** [review]
LGTM

File: changelog.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
   `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` to avoid a recursion issue.
 - Added `cai_causal_graph.causal_graph.CausalGraph.add_edges_from_paths` convenience method to the `cai_causal_graph.causal_graph.CausalGraph` class,
   in order to add edges from paths.
+- Fixed bug where use of `itertools.pairwise` would result in errors for Python \< 3.10.
```
Comment (andrew-causalens) [inline changelog.md:12]: No need as bug was never introduced as we never released 0.4.0 with it.

File: cai_causal_graph/utils.py [hunk 2]
```diff
@@ -124,3 +125,17 @@ def extract_names_and_lags(
...
+T = TypeVar('T')
+
+
+def pairwise(iterable: Iterable[T]) -> Iterable[Tuple[T, T]]:
+    """
+    Equivalent to `itertools.pairwise`.
+
+    This is used because `itertools.pairwise` method is not evailable for Python 3.8 and 3.9.
+    """
+    a, b = tee(iterable)
```
Comment (andrew-causalens) [inline cai_causal_graph/utils.py:139]: Suggest being explicit with `n=2`.

**Review by andrew-causalens — APPROVED** [review]
LGTM. Thanks for the quick fix. Just 2 minor comments then I will merge.

**Pull request merged**